snuba/datasets/plans/storage_processing.py

```diff
@@ -145,6 +172,21 @@ def apply_storage_processors(
                 storage_key=storage.get_storage_key(),
             )
         )
+
+    if _is_downsampled_storage_key(storage_key):
+        original_table = query_plan.query.get_from_clause()
+        query_plan.query.set_from_clause(
+            Table(
+                table_name=_get_corresponding_table(storage_key),
+                schema=original_table.schema,
+                storage_key=original_table.storage_key,
+                allocation_policies=original_table.allocation_policies,
+                final=original_table.final,
+                sampling_rate=original_table.sampling_rate,
+                mandatory_conditions=original_table.mandatory_conditions,
+            )
+        )
+
```

Comment:
Why is there modifications in the core query pipeline for this one storage selector?